iwyu.cc

```diff
@@ -1450,20 +1450,6 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
 
   set<const Type*> GetProvidedTypesForAlias(const Type* underlying_type,
```

Comment:
I wonder if it would make sense to remove this now that it's just an empty wrapper for `GetProvidedTypes`. Or would it make the patch more confusing?

Reply:
I'm not sure. I thought that leaving them keeps the code more consistent with the other cases (`GetProvidedTypesForFnReturn`, `GetProvidedTypesForAutocast`), and a good compiler should throw away redundant indirection...

Reply:
`FnReturn` and `Autocast` are all about the context, whereas `Alias` is just a quality of the type itself. I find it confusing now that `GetProvidedTypes` recurses via `GetProvidedTypesForAlias`.

But if there's a chance that we may want to tweak processing for alias types in the future, then I guess it makes sense to leave it.

Reply:
Done as a separate commit.